upgrade/upgrade-1.4.0.php

```diff
@@ -41,6 +41,131 @@ function upgrade_module_1_4_0($module)
         $result = $result && (bool) $meta->delete();
     }
 
+    // Fix multiple OrderState created in multishop before 1.3.0
+    $db = \Db::getInstance();
+
+    $queryConfigurationResults = $db->executeS('
+        SELECT c.id_configuration, c.name, c.value, c.id_shop, c.id_shop_group, os.id_order_state
+        FROM `' . _DB_PREFIX_ . 'configuration` AS c
+        LEFT JOIN `' . _DB_PREFIX_ . 'order_state` AS os ON (c.value = os.id_order_state)
+        WHERE c.name LIKE "PS_CHECKOUT_STATE_%"
+    ');
+
+    $orderStateRows = [];
+
+    if (false === empty($queryConfigurationResults)) {
+        foreach ($queryConfigurationResults as $queryConfigurationResult) {
+            if ($queryConfigurationResult['name'] === 'PS_CHECKOUT_STATE_WAITING_PAYPAL_PAYMENT') {
+                $orderStateRows['PS_CHECKOUT_STATE_WAITING_PAYPAL_PAYMENT'][] = [
+                    'id_configuration' => $queryConfigurationResult['id_configuration'],
+                    'id_order_state' => $queryConfigurationResult['id_order_state'],
+                ];
+            }
+
+            if ($queryConfigurationResult['name'] === 'PS_CHECKOUT_STATE_WAITING_CREDIT_CARD_PAYMENT') {
+                $orderStateRows['PS_CHECKOUT_STATE_WAITING_CREDIT_CARD_PAYMENT'][] = [
```

Comment:
To avoid duplicated code, you could filter the values and continue. This is also interesting here as we have the indented code going fa

```php
    $orderStatesToClean = [
        'PS_CHECKOUT_STATE_WAITING_PAYPAL_PAYMENT',
        'PS_CHECKOUT_STATE_WAITING_CREDIT_CARD_PAYMENT',
        'PS_CHECKOUT_STATE_WAITING_LOCAL_PAYMENT',
        'PS_CHECKOUT_STATE_AUTHORIZED',
        'PS_CHECKOUT_STATE_PARTIAL_REFUND',
        'PS_CHECKOUT_STATE_WAITING_CAPTURE',
    ];

    if (false === empty($queryConfigurationResults)) {
        foreach ($queryConfigurationResults as $queryConfigurationResult) {
            if (!in_array($queryConfigurationResult['name'], $orderStatesToClean)_{
                continue;
            }

            $orderStateRows[$queryConfigurationResult['name']][] = [
                'id_configuration' => $queryConfigurationResult['id_configuration'],
                'id_order_state' => $queryConfigurationResult['id_order_state'],
            ];
        }
    }
```

Reply:
Done